jax/_src/api.py

```diff
@@ -333,15 +330,78 @@ def jit(
     DeviceArray([   0,    1,  256, 6561], dtype=int32)
   """
   if FLAGS.experimental_cpp_jit and not config.jax_dynamic_shapes:
-    return _cpp_jit(fun, static_argnums, static_argnames, device, backend,
+    return _jit("cpp", fun, static_argnums, static_argnames, device, backend,
                     donate_argnums, inline, keep_unused)
   else:
-    return _python_jit(fun, static_argnums, static_argnames, device, backend,
+    return _jit("python", fun, static_argnums, static_argnames, device, backend,
                        donate_argnums, inline, keep_unused, abstracted_axes)
 
+def _jit(
+    jit_implementation: Literal["cpp", "python", None],
+    fun: Callable,
+    static_argnums: Union[int, Iterable[int], None] = None,
+    static_argnames: Union[str, Iterable[str], None] = None,
+    device: Optional[xc.Device] = None,
+    backend: Optional[str] = None,
+    donate_argnums: Union[int, Iterable[int]] = (),
+    inline: bool = False,
+    keep_unused: bool = False,
+    abstracted_axes: Optional[Any] = None,
+  ) -> stages.Wrapped:
+  # Implemements common logic between CPP and Python backends
+  _check_callable(fun)
+
+  # Coerce input
+  donate_argnums = _ensure_index_tuple(donate_argnums)
+
+  try:
+    sig = inspect.signature(fun)
+
+  except ValueError:
+    # Some built-in functions don't support signature.
+    # See: https://github.com/python/cpython/issues/73485
+    # In this case no validation is done
+    static_argnums = () if static_argnums is None else _ensure_index_tuple(static_argnums)
+    static_argnames = () if static_argnames is None else _ensure_str_tuple(static_argnames)
+
+  else:
+    # Infer argnums and argnames according to docstring
+    static_argnums, static_argnames = _infer_argnums_and_argnames(
+        sig, static_argnums, static_argnames)
+
+    # Validation
+    validate_argnums(sig, static_argnums, "static_argnums")
+    validate_argnums(sig, donate_argnums, "donate_argnums")
+
+    validate_argnames(sig, static_argnames, "static_argnames")
+
+  # Compensate for static argnums absorbing args
+  donate_argnums = rebase_donate_argnums(donate_argnums, static_argnums)
+
+  if jit_implementation == "cpp":
+    return _cpp_jit(fun, static_argnums=static_argnums, static_argnames=static_argnames,
+                    device=device, backend=backend,
+                    donate_argnums=donate_argnums, inline=inline, keep_unused=keep_unused)
+
+  elif jit_implementation == "python":
+    return _python_jit(fun, static_argnums=static_argnums, static_argnames=static_argnames,
+                       device=device, backend=backend, donate_argnums=donate_argnums,
+                       inline=inline, keep_unused=keep_unused, abstracted_axes=abstracted_axes)
+
+  # Used only in in testing.
```

Comment:
I'm not sure which tests use this case? I'd rather not add it.

Reply:
This has now been changed over. Instead of using `jit_type` for parameterised tests, I have now implemented a tuple of 'jitters' in `test_utils` to still be able to easily parameterise tests